open-subshare/open-subshare-json-contract.md

```diff
@@ -0,0 +1,128 @@
+# Open subshare JSON contract
+
+To get things started, the client will send the initial request.
+The first item in the `userCharList` is the `memberAccountId`
+
+NOTE: every state checks memoMode - if memoMode is true, we cannot proceed
+
+```json
+{
+  "rgState": "PRELOADDATA",
+  "powerOnFileName": "BANNO.NEWSUBCREATE.V1.POW",
+  "userChrList": [
+    {"id": 1, "value": ""},
+    {"id": 2, "value": ""},
+    {"id": 3, "value": ""},
+    {"id": 4, "value": ""},
+    {"id": 5, "value": ""}
+  ],
+  "userNumList": [],
+  "rgSession": 1
+}
+```
+
+To which the poweron should respond with:
+NOTE: if memoMode true, categories won't exist
+
+```json
+{
+  "memoMode": false,
+  "results": {
+    "canOpenSubShare": true,
```

Comment:
@jrobinson01 , Would the 'canOpenSubshare' element be returned by the PowerOn if there were an error condition being returned as well... as would be the case if there were no eligible shares which could be created?

{
	"memoMode": false,
	"results": {
		"canOpenSubShare": true,
		"errorCode": 503,
		"loggingErrorMessage": "No eligible shares available to be created"
	}
}

Reply:
I think we can stick to the format we came up with for wd-by-check. In this case:
```json
{
  "errorCode": 503,
  "loggingErrorMessage": "TRANPERFORM Error:+TRANERROR"
}
```
I'll start an Errors section in the doc